aea/skills/base.py

```diff
@@ -643,12 +643,19 @@ def from_config(
         assert (
             configuration.directory is not None
         ), "Configuration must be associated with a directory."
+
+        # we put the initialization here because some skill components
+        # might need some info from the skill
+        # (e.g. see https://github.com/fetchai/agents-aea/issues/1095)
+        skill_context = SkillContext() if skill_context is None else skill_context
+        skill = Skill(configuration, skill_context)
+        skill_context._skill = skill
```

Comment:
in the specific case of the skill `fetchai/tac_control_contract:0.1.0`, we need this because the `__init__` method of the model `Parameters` calls the setter of `context.is_active = False`, which requires `skill.skill_id` in order to print a debug message.